From 93e17e95fc3fdfa14ff4fa13b456f5847879027c Mon Sep 17 00:00:00 2001
From: Ken Brown <kbrown@cornell.edu>
Date: Sun, 26 Mar 2017 12:16:29 -0400
Subject: [PATCH 1/3] Enable building of shared libraries on Cygwin

* lib/Makefile.am (libmuaux_la_LDFLAGS):
* libmailutils/Makefile.am (libmailutils_la_LDFLAGS):
* libmu_auth/Makefile.am (libmu_auth_la_LDFLAGS):
* libmu_cpp/Makefile.am (libmu_cpp_la_LDFLAGS):
* libmu_dbm/Makefile.am (libmu_dbm_la_LDFLAGS):
* libmu_scm/Makefile.am (libmu_scm_la_LDFLAGS):
* libmu_sieve/Makefile.am (libmu_sieve_la_LDFLAGS):
* libproto/imap/Makefile.am (libmu_imap_la_LDFLAGS):
* libproto/maildir/Makefile.am (libmu_maildir_la_LDFLAGS):
* libproto/mailer/Makefile.am (libmu_mailer_la_LDFLAGS):
* libproto/mbox/Makefile.am (libmu_mbox_la_LDFLAGS):
* libproto/mh/Makefile.am (libmu_mh_la_LDFLAGS):
* libproto/nntp/Makefile.am (libmu_nntp_la_LDFLAGS):
* libproto/pop/Makefile.am (libmu_pop_la_LDFLAGS):
* python/libmu_py/Makefile.am (libmu_py_la_LDFLAGS):
Add -no-undefined.
---
 lib/Makefile.am              | 2 +-
 libmailutils/Makefile.am     | 2 +-
 libmu_auth/Makefile.am       | 2 +-
 libmu_cpp/Makefile.am        | 2 +-
 libmu_dbm/Makefile.am        | 2 +-
 libmu_scm/Makefile.am        | 2 +-
 libmu_sieve/Makefile.am      | 2 +-
 libproto/imap/Makefile.am    | 2 +-
 libproto/maildir/Makefile.am | 2 +-
 libproto/mailer/Makefile.am  | 2 +-
 libproto/mbox/Makefile.am    | 2 +-
 libproto/mh/Makefile.am      | 2 +-
 libproto/nntp/Makefile.am    | 2 +-
 libproto/pop/Makefile.am     | 2 +-
 python/libmu_py/Makefile.am  | 2 +-
 15 files changed, 15 insertions(+), 15 deletions(-)

diff --git a/lib/Makefile.am b/lib/Makefile.am
index 814e38a..94e9593 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -36,7 +36,7 @@ libmuaux_la_SOURCES = \
  mu_umaxtostr.c\
  mu_umaxtostr.h
 libmuaux_la_LIBADD=gnu/libgnu.la
-libmuaux_la_LDFLAGS = -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmuaux_la_LDFLAGS = -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 
 libmutcpwrap_a_SOURCES = tcpwrap.c
 
diff --git a/libmailutils/Makefile.am b/libmailutils/Makefile.am
index a105f38..a5927f8 100644
--- a/libmailutils/Makefile.am
+++ b/libmailutils/Makefile.am
@@ -51,5 +51,5 @@ libmailutils_la_LIBADD = \
  stdstream/libstdstream.la\
  url/liburl.la
 
-libmailutils_la_LDFLAGS = -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmailutils_la_LDFLAGS = -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 
diff --git a/libmu_auth/Makefile.am b/libmu_auth/Makefile.am
index b74d5cf..38afa7c 100644
--- a/libmu_auth/Makefile.am
+++ b/libmu_auth/Makefile.am
@@ -42,5 +42,5 @@ else
 endif
 
 libmu_auth_la_LIBADD = ${MU_LIB_MAILUTILS} @MU_AUTHLIBS@ @SQLLIB@ @LTLIBINTL@
-libmu_auth_la_LDFLAGS = -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmu_auth_la_LDFLAGS = -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 libmu_auth_la_DEPENDENCIES = @SQLLIB_DEPENDENCY@ 
diff --git a/libmu_cpp/Makefile.am b/libmu_cpp/Makefile.am
index 5e0dad1..f25684f 100644
--- a/libmu_cpp/Makefile.am
+++ b/libmu_cpp/Makefile.am
@@ -43,5 +43,5 @@ libmu_cpp_la_SOURCES = \
  url.cc
 
 libmu_cpp_la_LIBADD = @MU_COMMON_LIBRARIES@
-libmu_cpp_la_LDFLAGS = -rpath $(libdir)  -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmu_cpp_la_LDFLAGS = -no-undefined -rpath $(libdir)  -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 
diff --git a/libmu_dbm/Makefile.am b/libmu_dbm/Makefile.am
index 5ec6838..223a27e 100644
--- a/libmu_dbm/Makefile.am
+++ b/libmu_dbm/Makefile.am
@@ -43,5 +43,5 @@ libmu_dbm_la_SOURCES = \
 noinst_HEADERS = mudbm.h
 
 libmu_dbm_la_LIBADD = ${MU_LIB_MAILUTILS} @MU_AUTHLIBS@ @DBMLIBS@ @LTLIBINTL@
-libmu_dbm_la_LDFLAGS = -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmu_dbm_la_LDFLAGS = -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 libmu_dbm_la_DEPENDENCIES = @DBMLIB_DEPENDENCY@ 
diff --git a/libmu_scm/Makefile.am b/libmu_scm/Makefile.am
index b88f16e..032bbe8 100644
--- a/libmu_scm/Makefile.am
+++ b/libmu_scm/Makefile.am
@@ -39,7 +39,7 @@ libmu_scm_la_SOURCES=\
  $(C_SRCS) \
  mu_scm.h
 
-libmu_scm_la_LDFLAGS = -rpath $(libdir) -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmu_scm_la_LDFLAGS = -no-undefined -rpath $(libdir) -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 libmu_scm_la_LIBADD = \
  ${MU_LIB_MBOX}\
  ${MU_LIB_IMAP}\
diff --git a/libmu_sieve/Makefile.am b/libmu_sieve/Makefile.am
index 1e75e33..923003a 100644
--- a/libmu_sieve/Makefile.am
+++ b/libmu_sieve/Makefile.am
@@ -48,7 +48,7 @@ libmu_sieve_la_SOURCES = \
  util.c\
  variables.c
 libmu_sieve_la_LIBADD = ${MU_LIB_MAILUTILS} @LTDL_LIB@
-libmu_sieve_la_LDFLAGS = -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmu_sieve_la_LDFLAGS = -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 
 noinst_HEADERS = sieve-priv.h
 
diff --git a/libproto/imap/Makefile.am b/libproto/imap/Makefile.am
index 354e881..18f98df 100644
--- a/libproto/imap/Makefile.am
+++ b/libproto/imap/Makefile.am
@@ -18,7 +18,7 @@
 AM_CPPFLAGS = @MU_LIB_COMMON_INCLUDES@ 
 
 lib_LTLIBRARIES = libmu_imap.la
-libmu_imap_la_LDFLAGS=-version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmu_imap_la_LDFLAGS= -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 libmu_imap_la_LIBADD = ${MU_LIB_AUTH} ${MU_LIB_MAILUTILS} @INTLLIBS@ 
 
 SUBDIRS = . tests
diff --git a/libproto/maildir/Makefile.am b/libproto/maildir/Makefile.am
index da3a69c..bf6afd7 100644
--- a/libproto/maildir/Makefile.am
+++ b/libproto/maildir/Makefile.am
@@ -18,7 +18,7 @@
 AM_CPPFLAGS = @MU_LIB_COMMON_INCLUDES@ 
 
 lib_LTLIBRARIES = libmu_maildir.la
-libmu_maildir_la_LDFLAGS=-version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmu_maildir_la_LDFLAGS= -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 libmu_maildir_la_LIBADD = ${MU_LIB_MAILUTILS}
 libmu_maildir_la_SOURCES = \
  folder.c \
diff --git a/libproto/mailer/Makefile.am b/libproto/mailer/Makefile.am
index 0abda87..3bb8e49 100644
--- a/libproto/mailer/Makefile.am
+++ b/libproto/mailer/Makefile.am
@@ -17,7 +17,7 @@
 AM_CPPFLAGS = @MU_LIB_COMMON_INCLUDES@
 
 lib_LTLIBRARIES = libmu_mailer.la
-libmu_mailer_la_LDFLAGS=-version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmu_mailer_la_LDFLAGS= -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 libmu_mailer_la_LIBADD = ${MU_LIB_AUTH} ${MU_LIB_MAILUTILS} @INTLLIBS@
 
 if MU_COND_GSASL
diff --git a/libproto/mbox/Makefile.am b/libproto/mbox/Makefile.am
index 1c936bf..5bff219 100644
--- a/libproto/mbox/Makefile.am
+++ b/libproto/mbox/Makefile.am
@@ -18,7 +18,7 @@
 AM_CPPFLAGS = @MU_LIB_COMMON_INCLUDES@ 
 
 lib_LTLIBRARIES = libmu_mbox.la
-libmu_mbox_la_LDFLAGS=-version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmu_mbox_la_LDFLAGS= -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 libmu_mbox_la_LIBADD = ${MU_LIB_MAILUTILS}
 libmu_mbox_la_SOURCES = \
  folder.c\
diff --git a/libproto/mh/Makefile.am b/libproto/mh/Makefile.am
index 998540b..896b9ef 100644
--- a/libproto/mh/Makefile.am
+++ b/libproto/mh/Makefile.am
@@ -18,7 +18,7 @@
 AM_CPPFLAGS = @MU_LIB_COMMON_INCLUDES@
 
 lib_LTLIBRARIES = libmu_mh.la
-libmu_mh_la_LDFLAGS=-version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmu_mh_la_LDFLAGS= -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 libmu_mh_la_LIBADD = ${MU_LIB_MAILUTILS}
 libmu_mh_la_SOURCES = \
  folder.c\
diff --git a/libproto/nntp/Makefile.am b/libproto/nntp/Makefile.am
index 98c93c3..4df8054 100644
--- a/libproto/nntp/Makefile.am
+++ b/libproto/nntp/Makefile.am
@@ -18,7 +18,7 @@
 AM_CPPFLAGS = @MU_LIB_COMMON_INCLUDES@
 
 lib_LTLIBRARIES = libmu_nntp.la
-libmu_nntp_la_LDFLAGS=-version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmu_nntp_la_LDFLAGS= -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 libmu_nntp_la_LIBADD = ${MU_LIB_MAILUTILS}
 libmu_nntp_la_SOURCES = \
  nntp_article.c \
diff --git a/libproto/pop/Makefile.am b/libproto/pop/Makefile.am
index b69fbff..03516f7 100644
--- a/libproto/pop/Makefile.am
+++ b/libproto/pop/Makefile.am
@@ -18,7 +18,7 @@
 AM_CPPFLAGS = @MU_LIB_COMMON_INCLUDES@
 
 lib_LTLIBRARIES = libmu_pop.la
-libmu_pop_la_LDFLAGS=-version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
+libmu_pop_la_LDFLAGS= -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 libmu_pop_la_LIBADD = ${MU_LIB_AUTH} ${MU_LIB_MAILUTILS} @INTLLIBS@
 
 libmu_pop_la_SOURCES = \
--- a/python/2/libmu_py/Makefile.am	2018-02-10 01:06:00.000000000 -0500
+++ b/python/2/libmu_py/Makefile.am	2018-11-18 18:05:27.349230700 -0500
@@ -44,7 +44,7 @@ libmu_py_la_SOURCES = \
  registrar.c \
  url.c \
  util.c
-libmu_py_la_LDFLAGS = -rpath $(libdir) \
+libmu_py_la_LDFLAGS = -no-undefined -rpath $(libdir) \
  -export-symbols-regex '^(mu_py_|Py).*' \
  -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 libmu_py_la_LIBADD = $(PYTHON_LIBS) @MU_COMMON_LIBRARIES@ \
--- a/python/3/libmu_py/Makefile.am	2018-07-27 10:26:31.000000000 -0400
+++ b/python/3/libmu_py/Makefile.am	2018-11-18 18:06:01.223701700 -0500
@@ -44,7 +44,7 @@ libmu_py_la_SOURCES = \
  registrar.c \
  url.c \
  util.c
-libmu_py_la_LDFLAGS = -rpath $(libdir) \
+libmu_py_la_LDFLAGS = -no-undefined -rpath $(libdir) \
  -export-symbols-regex '^(mu_py_|Py).*' \
  -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 libmu_py_la_LIBADD = $(PYTHON_LIBS) @MU_COMMON_LIBRARIES@ \
